File: exercise_set/ruby/characters_to_string_solution.rb
def head(list)
  list[0]
end

def tail(list)
  list[1..]
end

def make_string(list)
  if list.size == 1
    head(list)
  else
    head(list) + make_string(tail(list))
  end
end

pet = ['T','i','n','y', ' ', 'C', 'a', 't']
puts make_string(pet) == "Tiny Cat" # test for correct return
puts pet == ['T','i','n','y', ' ', 'C', 'a', 't'] # test that pet array has not been mutated

File: exercise_set/javascript/characters_to_string_solution.js
function head(array) {
  return array[0];
}

function tail(array) {
  return array.slice(1);
}

function make_string(array) {
  if (array.length === 1) {
    return head(array);
  } else {
    return head(array) + make_string(tail(array));
  }
}

let pet = ['T','i','n','y', ' ', 'C', 'a', 't'];
console.log(make_string(pet) === "Tiny Cat");

